src/structures/Guild.ts

```diff
@@ -0,0 +1,679 @@
+    icon?: string | null;
+    ownerId?: Snowflake;
+    splash?: string | null;
+    banner?: string | null;
+    systemChannelId?: Snowflake | null;
+    rulesChannelId?: Snowflake | null;
+    publicUpdatesChannelId?: Snowflake | null;
+    preferredLocale?: string | null;
+  }, reason?: string): Promise<RestGuild> {
+    const guild = await this.client.rest.modifyGuild(this.id, {
+      name: options.name,
+      region: options.region,
+      verification_level: options.verificationLevel,
+      default_message_notifications:
+        typeof options.defaultNotifyAllMessages === "boolean"
+          ? +!options.defaultNotifyAllMessages as 0 | 1
+          : options.defaultNotifyAllMessages,
+      explicit_content_filter: options.explicitContentFilter,
+      afk_channel_id: options.afkChannelId,
+      afk_timeout: options.afkTimeout,
+      icon: options.icon,
+      owner_id: options.ownerId,
+      splash: options.splash,
+      banner: options.banner,
+      system_channel_id: options.systemChannelId,
+      rules_channel_id: options.rulesChannelId,
+      public_updates_channel_id: options.publicUpdatesChannelId,
+      preferred_locale: options.preferredLocale,
+    }, reason);
+
+    return new RestGuild(this.client, guild);
+  }
+
+  /** Creates a new channel. */
+  async createChannel(
+    name: string,
+    options?: CreateChannel & { type?: "text" },
+    reason?: string,
+  ): Promise<TextChannel>;
+  async createChannel(
+    name: string,
+    options?: CreateChannel & { type: "voice" },
+    reason?: string,
+  ): Promise<VoiceChannel>;
+  async createChannel(
+    name: string,
+    options?: CreateChannel & { type: "category" },
+    reason?: string,
+  ): Promise<CategoryChannel>;
+  async createChannel(
+    name: string,
+    options?: CreateChannel & { type: "news" },
+    reason?: string,
+  ): Promise<NewsChannel>;
+  async createChannel(
+    name: string,
+    options?: CreateChannel & { type: "store" },
+    reason?: string,
+  ): Promise<StoreChannel>;
+  async createChannel(
+    name: string,
+    options: CreateChannel = {},
+    reason?: string,
+  ): Promise<GuildChannels> {
+    const permissionOverwrites = options.permissionOverwrites?.map(
+      ({ permissions, id, type }) => {
+        const { allow, deny } = unparsePermissionOverwrite(permissions);
+
+        return {
+          id,
+          type,
+          allow,
+          deny,
+        };
+      },
+    );
+
+    const channel = await this.client.rest.createGuildChannel(this.id, {
+      name,
+      type: options.type ? channelTypeMap[options.type] : undefined,
+      topic: options.topic,
+      bitrate: options.bitrate,
+      user_limit: options.userLimit,
+      rate_limit_per_user: options.slowmode,
+      position: options.position,
+      permission_overwrites: permissionOverwrites,
+      parent_id: options.parentId,
+      nsfw: options.nsfw,
+    }, reason);
+
+    return this.client.newChannelSwitch(channel) as GuildChannels;
```

Comment:
Questionable cast? Consider using explicit return type instead to avoid mistakes.